src/components/book.vue

```diff
@@ -100,87 +100,20 @@ export default {
         }
     },
     methods: {
-        getBook(src,id){
-            switch (src){
-            case 'Goodreads':
-              //  this.$set(this,'result',this.GetbookGoodread(this.id)); //   
-              this.result = this.GetbookGoodread(id);
-                break;
-            case 'Google':
-                this.result = this.GetbookGoogle(id);
-                break;
-            default: 
-                this.$router.push('/');
-           }
+        getBook: function (src,id){
+          this.$store
+          .dispatch("grapBook", { 
+            src: src,
+            id : id
+           })
+          .then(res => {
+            this.result = this.$store.getters.grapBook;
+          })
+          .catch(err => {
+            console.log("Grap book Failed", err);
```

Comment:
Consider refactoring the code from grapBook to grabBook which is the correct spelling.

Reply:
yeah I felt that something was wrong, thanks :D